packages/channels/botpress-channel-telegram/src/telegram.js

```diff
@@ -76,6 +76,14 @@ class Telegram {
     })
   }
 
+  sendPhoto(chatId, photo, options) {
+    this.validateBeforeSending(chatId, options)
+
+    return Promise.fromCallback(() => {
```

Comment:
Why use fromCallback here? This promise will never ever resolve

Reply:
I replicated the behavior of the `sendText` function. If you go to the `sendPhoto()` method definition in the `node-telegram-bot-api` library, it returns a promise (Same as `sendText()`):

```js
  /**
   * Send photo
   * @param  {Number|String} chatId  Unique identifier for the message recipient
   * @param  {String|stream.Stream|Buffer} photo A file path or a Stream. Can
   * also be a `file_id` previously uploaded
   * @param  {Object} [options] Additional Telegram query options
   * @param  {Object} [fileOptions] Optional file related meta-data
   * @return {Promise}
   * @see https://core.telegram.org/bots/api#sendphoto
   * @see https://github.com/yagop/node-telegram-bot-api/blob/master/doc/usage.md#sending-files
   */
  sendPhoto(chatId, photo, options = {}, fileOptions = {}) {}
```

I'm not sure why `Promise.fromCallback` is used in both methods. I think it might be unnecessary, since the library already returns a promise. Maybe it was used to ensure that the class methods always return a promise.

Reply:
The same happens with the slack client:

packages/channels/botpress-channel-slack/src/slack.js
```js
  sendText(channelId, text, options) {
    this.validateBeforeSending(channelId, options)
    this.validateText(text)

    return Promise.fromCallback(cb => {
      this.web.chat.postMessage(channelId, text, options, cb)
    })
  }
```

`postMessage` method [also returns a promise](https://slackapi.github.io/node-slack-sdk/web_api#posting-a-message).

Any idea?

Reply:
@slvnperron In fact, I believe that any promise of any channel *will never ever resolve* if this workaround is used across the channels.

Reply:
Hi. You were right, the promises were never resolved due to the use of the `fromCallback` function. I have removed that function calls to return the original promise.

As I pointed out in my last comment, this error may be occurring in other channels (e.g. **slack**). Could you check it?

I'll try to code some improvements to the telegram channel soon. **You can merge this branch whenever you want**.